lib/Core/ExecutionState.cpp

```diff
@@ -34,6 +34,9 @@ using namespace klee;
 namespace { 
   cl::opt<bool>
   DebugLogStateMerge("debug-log-state-merge");
+
+  // HACK: This is really not good for library design
+  uint64_t globalExecutionStateCounter = 0;
```

Comment:
That doesn't really fix the problem. The problem is that having a global counter means KLEE can't be effectively used as a library. Imagine for example running two executors in parallel in different threads and running them sequentially.

What we really need is some sort of `KLEEContext` object that we can store things like this in to allow better re-use of KLEE's components. There are lots examples of this problem in other parts of KLEE's code so this PR is probably not the right time to fix it.